distributed/submit/remote_cli.py

```diff
@@ -0,0 +1,34 @@
+import socket
+import logging
+
+import click
+
+from tornado.ioloop import IOLoop
+from distributed.submit.remote_client import RemoteClient
+
+from distributed.utils import get_ip
+
+logger = logging.getLogger('distributed.remote')
+
+
+@click.command()
+@click.option('--host', type=str, default=None,
+              help="IP or hostname of this server")
+@click.option('--port', type=int, default=8788, help="Remote Client Port")
+def remote(host, port):
+    _remote(host, port)
+
+
+
+def _remote(host, port, loop=IOLoop.current(), client=RemoteClient):
+    host = host or get_ip()
+    if ':' in host and port == 8788:
+        host, port = host.rsplit(':', 1)
+        port = int(port)
+    ip = socket.gethostbyname(host)
+    remote_client = client(ip=ip, loop=loop)
+    remote_client.start(port=port)
+    loop.start()
+    loop.close()
+    remote_client.stop()
+    logger.info("End remote client at %s:%d", host, port)
```

Comment:
I apologize, my comments from before that it's nice to bring out a lot of logic from the CLI module weren't entirely clear.  

What I meant to say was that it's nice to have tests that don't rely on cli technology, like click and opening processes, but instead rely on the code that those cli modules use.  In the case of submit I suspect that what I'm suggesting is that we create tests based on `RemoteClient` and the `_submit` coroutine, not that we bring the click stuff out from the cli directory into other directories.  Click stuff should definitely stay in CLI.  Starting and stopping IOLoops should probably stay there as well.  Ideally we find a way to test most of the logic of our code without using these technologies; that way we can rely on the testing framework to set up and close down IOLoops and processes for us cleanly.  This should reduce testing frustration.